src/server/pfs/server/master.go

```diff
@@ -31,20 +33,24 @@ const (
 
 func (d *driver) master(ctx context.Context) {
 	ctx = auth.AsInternalUser(ctx, "pfs-master")
-	masterLock := dlock.NewDLock(d.etcdClient, path.Join(d.prefix, masterLockPath))
 	backoff.RetryUntilCancel(ctx, func() error { //nolint:errcheck
-		masterCtx, err := masterLock.Lock(ctx)
-		if err != nil {
-			return errors.EnsureStack(err)
-		}
-		defer masterLock.Unlock(masterCtx) //nolint:errcheck
-		eg, ctx := errgroup.WithContext(masterCtx)
+		eg, ctx := errgroup.WithContext(ctx)
 		trackerPeriod := time.Second * time.Duration(d.env.StorageConfig.StorageGCPeriod)
 		if trackerPeriod <= 0 {
 			d.log.Info("Skipping Storage GC")
 		} else {
-			d.log.Infof("Starting Storage GC with period=%v", trackerPeriod)
 			eg.Go(func() error {
+				lock := dlock.NewDLock(d.etcdClient, path.Join(d.prefix, masterLockPath, "gc"))
+				d.log.Infof("Starting Storage GC with period=%v", trackerPeriod)
+				gcCtx, err := lock.Lock(ctx)
```

Comment:
The gc run call isn't receiving this context. A common pattern with contexts is to overwrite `ctx` throughout a function so it is always handled correctly, or create a locally scoped `ctx` if you want to keep the outer `ctx` the same. In this case a locally scoped `ctx` would work, so change this to `ctx, err := ...`.

Reply:
Oh I totally missed, that good catch!. I'll just pass `gcCtx` to `gc.RunForever(gcCtx)`